contracts/ethereum/src/PaymentRegistry.sol

```diff
@@ -56,67 +49,55 @@ contract PaymentRegistry is Initializable, OwnableUpgradeable, UUPSUpgradeable {
-        // 64900 gas:
-        transfers[index].destAddress = destAddress;
-        transfers[index].amount = msg.value;
-        transfers[index].isUsed = true;
-        transfers[index].chainId = chainId;
-
 
-        (bool success,) = payable(address(uint160(destAddress))).call{value: msg.value}(""); //34000 gas
-        // (bool success,) = payable(marketMaker).call{value: msg.value}(""); //32000 gas //to implement this, address must be changed to from uint256 to addr
+        (bool success,) = payable(destAddress).call{value: msg.value}(""); //34000 gas
 
         require(success, "Transfer failed.");
-        emit Transfer(orderId, msg.sender, transfers[index]); //3000 gas
+        emit Transfer(orderId, msg.sender, destAddress, msg.value, chainId); //3000 gas (todo verify gas)
     }
 
-//TODO change name to claimPaymentStarknet
-    function claimPayment(uint256 orderId, uint256 destAddress, uint256 amount) external payable onlyOwnerOrMM {
+    function claimPayment(uint256 orderId, address destAddress, uint256 amount) external payable onlyOwnerOrMM {
         bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, amount, Chain.Starknet));
-        TransferInfo storage transferInfo = transfers[index];
-        require(transferInfo.isUsed == true, "Transfer not found.");
-
-        uint256[] memory payload = new uint256[](5); //TODO why array of 256 if then filled with 128?
-        payload[0] = uint128(orderId); // low
-        payload[1] = uint128(orderId >> 128); // high
-        payload[2] = transferInfo.destAddress;
-        payload[3] = uint128(amount); // low
-        payload[4] = uint128(amount >> 128); // high
-            
+        require(transfers[index] == true, "Transfer not found.");
+
+        //changed now only uint256[3];
+        uint256[] memory payload = new uint256[](3);
+        payload[0] = uint256(orderId);
+        payload[1] = uint256(uint160(destAddress));
+        payload[2] = uint256(amount);
+        
         _snMessaging.sendMessageToL2{value: msg.value}(
             StarknetEscrowAddress,
             StarknetEscrowClaimPaymentSelector,
             payload);
-
-        emit ClaimPayment(transferInfo);
     }
 
-// why is destAddress a uint256? should be address?
-// oh, because sn address is not "address" typew]
     function claimPaymentZKSync(
-        uint256 orderId, uint256 destAddress, uint256 amount,
+        uint256 orderId, address destAddress, uint256 amount,
         uint256 gasLimit,
         uint256 gasPerPubdataByteLimit
     ) external payable onlyOwnerOrMM {
         bytes32 index = keccak256(abi.encodePacked(orderId, destAddress, amount, Chain.ZKSync));
-        TransferInfo storage transferInfo = transfers[index];
-        require(transferInfo.isUsed == true, "Transfer not found.");
+        require(transfers[index] == true, "Transfer not found.");
+        transfers[index] = false; //this transfer has been claimed
```

Comment:
This method should not set transfer[index] = false. Maybe it could have two arrays. One with transfer and the another one with claims

Reply:
Agreed, i will change the bool to a byte, and use the bits inside the byte to check this values. This will also leave an extra 6 bits to use in the future if needed

Reply:
done 👍